eth/catalyst/simulated_beacon.go

```diff
@@ -159,8 +159,11 @@ func (c *SimulatedBeacon) sealBlock(withdrawals []*types.Withdrawal, timestamp u
 
 	// Reset to CurrentBlock in case of the chain was rewound
 	if header := c.eth.BlockChain().CurrentBlock(); c.curForkchoiceState.HeadBlockHash != header.Hash() {
-		finalizedHash := c.finalizedBlockHash(header.Number.Uint64())
-		c.setCurrentState(header.Hash(), *finalizedHash)
+		if fh := c.finalizedBlockHash(header.Number.Uint64()); fh == nil {
```

Comment:
Why doesn't this break in upstream?

Reply:
Actually i caught this in nitro test https://github.com/OffchainLabs/nitro/actions/runs/13568710873/job/37927853760, so probably the geth tests are not extensive enough. (Also on nitro side as well its falky only.)

Reply:
Okay. Thanks.